doc/dune-files.rst

```diff
@@ -437,6 +437,29 @@ Starting with Dune 2.0, Dune mangles compilation units of executables by
 default. However, this can still be turned off using ``(wrapped_executables
 false)``
 
+.. _map-workspace-root:
+
+map_workspace_root
+-------------------
+
+It is desirable that the output of tools not contain references to the
+file system location from which they were built. Starting from Dune 3.6,
+Dune has mapped references to the workspace directory to "/workspace_root".
+
+Starting from Dune 3.7, an option is available to turn on/off mapping
+of the workspace on a per-project basis:
+
+.. code:: scheme
+
+    (map_workspace_root <bool>)
+
+Starting with Dune 3.0, Dune does this mapping by default. However,
```

Comment:
Should this be 3.6?

Reply:
(Also, it seems like there is some duplication between this paragraph and the one above.)

Reply:
> Should this be 3.6?

I was basing 3.0 on the following code:

```
let add_workspace_root_to_build_path_prefix_map t = t.dune_version >= (3, 0)
```
 which is the existing (before this PR) test for whether to map or not.

Actually, I see some ambiguity. There is the version of Dune itself, and there is the version as specified in the "(lang dune ...)" stanza in the dune-project file. Do we document the Dune version or the Dune language version? Or are they supposed to be consistent? If they are supposed to be consistent, then the above would imply that this mapping started at version 3.0 (I don't know the actual history). 

Currently this PR, in the absence of a user-specified "map_workspace_root" stanza, uses the default given by:

```
let map_workspace_root_default ~(lang : Lang.Instance.t) =
  lang.version >= (3, 6)
```

which is not consistent with the behavior before this PR. So it seems this "3, 6" should be "3, 0". Do you agree?

Reply:
> (Also, it seems like there is some duplication between this paragraph and the one above.)

I have pushed a change to remove the redundancy.